src/content/app/tools/blast/state/blast-form/blastFormSlice.ts

```diff
@@ -97,16 +101,109 @@ const getDatabaseSequenceType = ({
+) => {
+  if (
+    state.sequences.length &&
+    state.settings.sequenceSelectionMode === 'automatic'
+  ) {
+    const firstSequence = state.sequences[0].value;
+    const sequenceType = guessSequenceType(firstSequence); // FIXME — should this only be done if there were no sequences before?
+    state.settings.sequenceType = sequenceType;
+  }
+
+  if (state.settings.databaseSelectionMode === 'automatic') {
+    updateDatabaseFromSequenceType(state, config);
+  }
+
+  // Update computed properties
+  const sequenceType = state.settings.sequenceType;
+  const database = state.settings.parameters.database as string;
+  const databaseSequenceType = getDatabaseSequenceType({
+    database,
+    config
+  });
+
+  const programName = getProgamName({
+    sequenceType,
+    databaseSequenceType,
+    config
+  });
+
+  const presetName = initialBlastFormSettings.preset;
+  const parameters = config.presets.settings[programName][presetName];
+
+  state.settings.program = programName;
+  state.settings.preset = presetName;
+  state.settings.parameters = { ...parameters, database };
+};
+
+const resetSettings = (state: BlastFormState, config: BlastSettingsConfig) => {
+  state.settings = cloneDeep(initialBlastFormSettings);
+  state.settings.parameters.database = config.defaults.database;
+};
+
+const updateDatabaseFromSequenceType = (
+  state: BlastFormState,
+  config: BlastSettingsConfig
+) => {
+  const currentDatabase = state.settings.parameters.database;
+  const currentSequenceType = state.settings.sequenceType;
+  const { database_sequence_types } = config;
+  if (
+    !currentDatabase ||
+    database_sequence_types[currentDatabase] !== currentSequenceType
+  ) {
+    // get first database with the matching sequence type
+    const match = Object.entries(config.database_sequence_types).find(
+      ([, dbSequenceType]) => dbSequenceType === currentSequenceType
+    ) as [string, SequenceType];
+    const [databaseName] = match;
+    state.settings.parameters.database = databaseName;
+  }
+};
+
+const updateEmptyInputVisibility = (
+  state: BlastFormState,
+  sequences: ParsedInputSequence[]
+) => {
+  if (sequences.length > state.sequences.length) {
+    state.shouldAppendEmptyInput = false;
+  } else if (!sequences.length) {
+    state.shouldAppendEmptyInput = true;
+  }
+};
+
 const blastFormSlice = createSlice({
   name: 'blast-form',
   initialState,
   reducers: {
     setSequences(
       state,
-      action: PayloadAction<{ sequences: ParsedInputSequence[] }>
+      action: PayloadAction<{
+        sequences: ParsedInputSequence[];
+        config: BlastSettingsConfig;
+      }>
     ) {
-      const { sequences } = action.payload;
+      const { sequences, config } = action.payload;
+      updateEmptyInputVisibility(state, sequences);
+
       state.sequences = sequences;
+
+      if (!sequences.length) {
+        resetSettings(state, config);
+        autoUpdateSettings(state, config); // FIXME is this required?
```

Comment:
it does feel its not required since the resetSettings have been updated. Did you try removing it and see if it breaks anything?

Reply:
But we still need to set the individual parameters based on the sequence type, database, and sensitivity preset. This code is currently in `autoUpdateSettings`, and it isn't clear to me whether it can safely be extracted in a dedicated function.

I'll keep the `autoUpdateSettings`, however silly it may look 😞